src/components/InventoryDetail/InventoryDetail.js

```diff
@@ -41,7 +42,16 @@ const InventoryDetail = ({
     useEffect(() => {
         const currId = inventoryId || location.pathname.replace(/\/$/, '').split('/').pop();
         if (!entity || !(entity?.id === currId) || !loaded) {
-            dispatch(loadEntity(currId, { hasItems: true }, { showTags }));
+            const action = loadEntity(currId, { hasItems: true }, { showTags });
+            dispatch(action);
+            action.payload.then(() => {
+                getEntitySystemProfile(currId).then(result => {
+                    const cloudProviderObj = (typeof result.results[0].system_profile.cloud_provider !== 'undefined')
+                        && result.results[0].system_profile.cloud_provider;
+                    dispatch(setRosTabVisibility((cloudProviderObj === 'aws' || cloudProviderObj === 'azure')));
+                    return result;
+                });
+            });
```

Comment:
I'd change the flow of action completely, instead of fetching the data and dispatching specific action for ROS, I'd disable the ROS tab by default. And I'd add new reducer that reacts to `ACTION_TYPES.LOAD_SYSTEM_PROFILE` and enables the ROS tab. `resourceOptTabVisibility` should be enough. This way we won't add additional logic to the shared InventoryDetail and at the same time make all custom tab hiding just for inventory app.

Approach in this PR (waiting for `getEntitySystemProfile`) could mean that ROS tab would not be visible even in ROS application itself, which is unwanted.

Reply:
> I'd disable the ROS tab by default.

@karelhala  That is what happens.

> And I'd add a new reducer that reacts to ACTION_TYPES.LOAD_SYSTEM_PROFILE and enables the ROS tab.
Using the action type `ACTION_TYPES.LOAD_SYSTEM_PROFILE` can result in more race conditions when switching between detail pages :disappointed:. Ideally, we would use the `thunk` middleware to chain the actions directly through redux though. Optimal way would be if the data was provided in the entity detail but that is not happening.

Reply:
The ROS tab should be disabled only on full inventory view. We can wait for the `ACTION_TYPES.LOAD_SYSTEM_PROFILE` (no need to fire a new one as it should already be fired) to load and enable the ROS tab in inventory app redux store and enable the tab if criterium is met. The problem I have with the approach in this PR is that we're adding extra logic to shared component just to disable the tab in inventory app.

Reply:
In that case, shouldn't his be handled by the ROS tab itself then?

Reply:
It's partially handleded by the ROS tab, they currently show empty state. But they don't want to show the tab at all for some users (I was against it, but the PM specifically required it)

Reply:
Well, I would personally move the responsibility to the team. As I said, provide a component API to specifically state which tabs to show. They will have full control and we have one thing less to worry about. I assume we will get requests for other tabs with similar behaviour in the future. I think it may come in handy when you navigate through the URL. This can cause some wire behaviour when you navigate directly through the URL, and your tab is missing for some reason and pops up later.